spec/api.yaml

```diff
@@ -124,6 +140,39 @@ paths:
               schema:
                 $ref: '#/components/schemas/User'
 
+  /user/edit-profile:
+       put:
+         summary: 'edit user details'
+         tags:
+           -user
+         parameters:
+           -$ref: '#/components/parameters/AuthToken'
+         operationId: EditUserProfile
+         requestBody: 
+            content:
+             application/json:
+               schema:
+                 $ref: '#/components/schemas/SignUpUserInfoRequest' 
```

Comment:
Don't use `SignUpUserInfoRequest` as it contains password, create a new schema that only takes in the name that the user wants to change to.